importer/likutim/compare_files.go

```diff
@@ -0,0 +1,280 @@
+				log.Errorf("Error on decode response . Error: %s", err)
+				e := make([]interface{}, 2)
+				e[0] = f
+				e[1] = err
+				c.errors = append(c.errors, e)
+				continue
+			}
+			log.Debug("Successfully count words.")
+			c.wordsByFileUid[f.UID] = ps
+			c.allDocs = append(c.allDocs, f.UID)
+		}
+	}
+}
+
+func (c *Compare) clearDuplicates(docs []string) {
+	log.Debugf("Start recursive function docs length: %d ", len(docs))
+	if len(docs) == 0 {
+		return
+	}
+
+	d := Double{
+		Save:    docs[0],
+		Doubles: []string{docs[0]},
+	}
+	var nextDocs []string
+
+	for i, n := range docs {
+		if i == 0 {
+			continue
+		}
+		eq, b := c.compareFiles(docs[0], n)
+		if eq {
+			d.Save = b
+			d.Doubles = append(d.Doubles, n)
+		} else {
+			nextDocs = append(nextDocs, n)
+		}
+	}
+	c.result = append(c.result, &d)
+	log.Debugf("Call recursive function docs length: %d", len(nextDocs))
+	c.clearDuplicates(nextDocs)
+}
+
+func (c *Compare) compareFiles(n1, n2 string) (bool, string) {
+	f1 := c.wordsByFileUid[n1]
+	f2 := c.wordsByFileUid[n2]
+	biggerName := n1
+
+	if len(f2) > len(f1) {
+		f1 = c.wordsByFileUid[n2]
+		f2 = c.wordsByFileUid[n1]
+		biggerName = n2
+	}
+
+	eq := 0
+	for _, p1 := range f1 {
+		hasP1 := false
+		for _, p2 := range f2 {
+			if c.compareParagraph(p1, p2) {
+				hasP1 = true
+				break
+			}
+		}
+
+		if hasP1 {
+			eq = eq + 1
+		}
+	}
+
+	if eq == len(f2) {
+		return true, biggerName
+	}
+	percent := float64(eq) / float64(len(f2))
+
+	return percent > 0.9, biggerName
+}
+
+func (c *Compare) compareParagraph(p1, p2 map[string]int64) bool {
+	keys := map[string]float64{}
+	wordsCount := float64(0)
+	for k, c := range p1 {
+		cf := float64(c)
+		keys[k] = cf
+		wordsCount += cf
+	}
+
+	for k, c := range p2 {
+		cf := float64(c)
+		wordsCount += cf
+		v, ok := keys[k]
+		if !ok {
+			keys[k] = cf
+			continue
+		}
+		diff := cf - v
+		if diff > 0 {
+			keys[k] = diff
+		} else {
+			keys[k] = -1 * diff
+		}
+	}
+
+	diffCount := float64(0)
+	for _, v := range keys {
+		diffCount += v
+	}
+	if wordsCount == 0 {
+		return false
+	}
+
+	return diffCount/wordsCount < 0.1
+}
+
+func (c *Compare) openDB() *sql.DB {
+	mdb, err := sql.Open("postgres", viper.GetString("mdb.url"))
+	utils.Must(err)
+	utils.Must(mdb.Ping())
+	boil.SetDB(mdb)
+	boil.DebugMode = true
+	utils.Must(common.InitTypeRegistries(mdb))
+	return mdb
+}
+
+func (c *Compare) printToCSV() {
+	addToFile("File that stay, All duplicates")
+	for _, r := range c.result {
+		addToFile(fmt.Sprintf(",\n%s, %v", r.Save, r.Doubles))
+	}
+	addToFile(",\nFiles  with exceptions")
+	for _, e := range c.errors {
+		addToFile(fmt.Sprintf(",\n%v, %v", e[0], e[1]))
+	}
+}
+
+func addToFile(line string) {
+	p := path.Join(viper.GetString("likutim.os-dir"), "kitvei-makor-duplicates.csv")
+	f, err := os.OpenFile(p, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	utils.Must(err)
+
+	if _, err := f.Write([]byte(line)); err != nil {
+		f.Close()
+		log.Error(err)
```

Comment:
Again, you handle error in wrong place.
If an error occurred while writing to file it's 99.99% likely to happen again and again for each row. So you probably want to stop trying.
But you don't return the error so your caller can't break his loop...